src/pages/home/home.ts

```diff
@@ -25,319 +25,347 @@ query {
-			var schema = await this.http.get(deskbookPath).toPromise();
-                        this.schema = schema;
-		} else {
-			var user = await this.auth.currentUser();
-			var files = [];
-
-			for (let file of user.jsonFiles){
+      for (let file of user.jsonFiles) {
         console.log(file);
-				var newFile = JSON.parse(file);
-        if (typeof newFile == 'string') {
+        var newFile = JSON.parse(file);
+        if (typeof newFile == "string") {
           newFile = JSON.parse(newFile);
         }
-				files.push(newFile);
-			}
+        files.push(newFile);
+      }
 
-			var deskbookFile = files.filter(f => f.fileName == deskbook);
-			this.schema = deskbookFile[0].file;
-		}
-	}
+      var deskbookFile = files.filter((f) => f.fileName == deskbook);
+      this.schema = deskbookFile[0].file;
+    }
+  }
 
-	////////// METHODS TO LAUNCH POPOVERS //////////////////////////////
-	// TODO:  abstract general popover logic<01-08-18, mpf> //
+  ////////// METHODS TO LAUNCH POPOVERS //////////////////////////////
+  // TODO:  abstract general popover logic<01-08-18, mpf> //
 
-    handleResetClick() {
-      this.popOver.create( PasswordResetComponent, {}, {cssClass: 'password-reset'})
-                    .present();
-    }
+  handleResetClick() {
+    this.popOver
+      .create(PasswordResetComponent, {}, { cssClass: "password-reset" })
+      .present();
+  }
 
-	showRegisterForm = () => this.showRegister = true;
-	mobileRegisterForm() {
-		this.showRegister = true;
-		this.mobileRegister = true;
-	}
+  showRegisterForm = () => (this.showRegister = true);
+  mobileRegisterForm() {
+    this.showRegister = true;
+    this.mobileRegister = true;
+  }
 
-	showThreads(myEvent) {
+  showThreads(myEvent) {
     this.threadsShown = !this.threadsShown;
-	}
+  }
 
-  addMember(nameIn:string,emailIn:string,roleIn:string){
-    var newMember = {name: nameIn, email: emailIn, role: roleIn};
+  addMember(nameIn: string, emailIn: string, roleIn: string) {
+		var newMember = { name: nameIn, email: emailIn, role: roleIn };
+		if ( this.validMemberInput(emailIn) ) {
     this.members.push(newMember);
     this.assForm.teamMembers.push(newMember);
 
-		var name = <any>(document.getElementById("memName"));
-		name.value = "";
-		var email = <any>(document.getElementById("memEmail"));
-		email.value = "";
-		var role = <any>(document.getElementById("memRole"));
-		role.value = "";
-		this.presentToast();
+    var name = <any>document.getElementById("memName");
+    name.value = "";
+    var email = <any>document.getElementById("memEmail");
+    email.value = "";
+    var role = <any>document.getElementById("memRole");
+    role.value = "";
+    this.presentToast();
+	};
+}
+
+  // all that is required is an email
+  // space this correctly!!!!
+  validMemberInput(emailInput) {
+    if (!emailInput) {
+      this.errors = ["no-email"];
+      return false;
+    } else if (
+      this.assessment.teamMembers.map((a) => a.email).includes(emailInput)
```

Comment:
explicit naming on the 'a' here to explain what it is

Reply:
This is not my code. Most of it is from previous bodies working in this code repo. I would enjoy tidying up the formatting and variable names. Maybe it's worth a Jira ticket?

Reply:
I see, sure you can suggest it. 

I suggest that you review your commits and make sure to commit only changes related to your feature: 

https://github.com/EktaFlow/docent-base/pull/179/commits/8e96703bb14618627660510506f5ca9267f2e3c4

Reply:
Changed